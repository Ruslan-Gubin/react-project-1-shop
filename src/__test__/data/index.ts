export {testMineArray} from './testMineArray';
export {responseTestData} from './getDataResponse';



