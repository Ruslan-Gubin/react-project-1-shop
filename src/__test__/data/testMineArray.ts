import { MineType } from "models/GameType"

const testMineArray: MineType[] = [
  {_id: '235klhsffrsd', resouces: 'iron',level: 11,imag: 'string',title: 'iron',piple: 1,income: 13},
  {_id: '235sdfklhfsd', resouces: 'iron',level: 15,imag: 'string',title: 'iron',piple: 5,income: 12},
  {_id: '235afklhffsd', resouces: 'wheat',level: 11,imag: 'string',title: 'iron',piple: 13,income: 11},
  {_id: '235klggfhfsd', resouces: 'wheat',level: 6,imag: 'string',title: 'iron',piple: 5,income: 6},
  {_id: '235klggfhfsd', resouces: 'wood',level: 6,imag: 'string',title: 'iron',piple: 5,income: 6},
]

export {testMineArray}