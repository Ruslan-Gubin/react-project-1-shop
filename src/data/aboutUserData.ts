const aboutUserData = {
  name: "Ruslan Gubin",
  text: "Привет, я веб-разработчик, с середины мая 2022 я начал изучать програмирование, первые месяцы обучения были посвящены css, html БЭМ-методология правильное построение компонентов блоков, было много создано шпаргалок и одностраничный сайтов по обучающим роликам, работал с прикольными анимациями. в июне сделал полноценный портфолио сайт где впервые внедрил технологию javascript, которая отвечал за отображение перевод на английский, swiper менял изображение в начале сайта, и плавный переход по странице, на то время сборщик проекта был gulp, gulp - конвертировал изображение на webp ели у пользователя был подходящий браузер, работал со шрифтами, в общем сжимал все до мелочей. После портфолио я посвятил все время изучению javascript, также были небольшие обучающиеся проекты, 15 июля сборщик webpack и свой первый js проект интернет магазин на нативном js, это был ооп подход, также параллельно собрал сервер на Node.js где взаимодействовал с сайтом, там уже можно было создать товар, удалить, получить отфильтрованный массив по поиску, добавление комментария жалобы и удаление,  уже поинтересней. В августе  начал создавать этот сайт. Это мой первый проект на React , тут я уже описал в отделе ниже все технологии, пришлось разобраться с backend поглубже и добавилось технологий frontenda, в соседнем отделе я опишу  возможности сайта,  хотелось бы еще добавить в будущем нормальную админку и еще много идей.  На данный момент ищу работу удаленно,  хотелось бы получить опыт работы с командой, опытных менторов, интересные задачи и обучатся новому.",
  test: "впервые внедрил технологию javascript, впервые внедрпервые внедрил технологию javascript, ",
};

const aboutSitePosts = [
  'Авторизация пользователя',
  'Вход пользователя',
  'Изменение пользователя',
  'Удаление пользователя',
  'Добавление и удаление постов',
  'Изменение постов',
  'Счетчик просмотров постов',
  'Лайк постов и комментариев',
  'Дизлайк постов и комментариев',
  'Просмотр количество лайков',
  'Добавление коментрия к посту',
  'Удаление коментрия к посту',
  'Изменение коментрия к посту',
  'Отображение и переход на страницу коментария',
  'Добавление пользователя в друзья',
  'Удаление пользователя из друзей',
  'Отказ пользователя в друзья',
  'Ожидание пользователя в друзья',
  'Диалог между пользователями',
  'Удаление диалога между пользователями',
  'Поиск по имени пользователя',
  'Поиск по заголовку поста',
  'Фильтрация поиска личных постов',
  'Фильтрация и отображение выделенного тега',
]

const aboutSiteProduct = [
  'Переход в отдел товаров',
  'Фильтрация товара по категориям',
  'Пагинация',
  'Добавление Товара до 5 изображений',
  'Замена и изменение товара и изображение',
  'Удаление товара',
  'Добавление в корзину',
  'Проверка количество товара перед покупкой',
  'Создание удаление комментария товара',
  'Заполнение Visa и обработка заказа',
]

export { aboutUserData, aboutSitePosts, aboutSiteProduct};
