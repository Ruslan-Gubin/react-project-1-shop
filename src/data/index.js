export { productSortingArray } from "./productSortingArray";
export { imagesSwiper } from "./imagesSwiper";
export { productsCategoriLink } from "./productsCategoriLink";

export {
  imagesUrlVk,
  imagesUrlFacebook,
  imagesUrlTwitter,
  imagesUrlInstagram,
} from './imagesUrl.ts';

export {
  eyeIcon,
  commentsIcon,
  userLogin,
  cartPng,
  removeGoodsPng,
  checkIcon,
  search,
  close,
  arrowLeft,
  userRegistedPng,
  outIcon,
} from "./icons";

export {
  categoryPosts,
  tegsArray,
  sortCategoryName,
  commentsArray,
} from './tegsAndComments';
