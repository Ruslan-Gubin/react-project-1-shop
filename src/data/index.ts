export { productSortingArray } from "./productSortingArray";
export { imagesSwiper } from "./imagesSwiper";
export { productsCategoriLink } from "./productsCategoriLink";
export {descriptionFrontend, descriptionBackend} from './descriptionOfTechnologies';
export {aboutUserData, aboutSitePosts, aboutSiteProduct} from './aboutUserData';
export {iconsGame} from './iconsGame';
export {resurceUpdateRuls} from './updateRules';

export {
  imagesUrlVk,
  imagesUrlFacebook,
  imagesUrlTwitter,
  imagesUrlInstagram,
} from './imagesUrl';

export {icons} from './icons';

export {
  categoryPosts,
  tegsArray,
  sortCategoryName,
} from './tegsAndComments';
