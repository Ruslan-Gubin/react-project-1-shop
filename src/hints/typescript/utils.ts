export {}

interface User {
    name: string
}

const user: Readonly<User> = { // все свойства Толко для чтения
    name: 'Ruslan'
}
//  Required<Props> // все поля обязвтельные

// Record<K, T> //
interface PageInfo {
    title: string
}

type Page = 'home' | 'about' | 'contact'

const x: Record<Page, PageInfo> = {
    about: {title: 'about'},
    contact: {title: 'contact'},
    home: {title: 'home'}
}
// Pick<>
interface Todo {
    title: string;
    description: string;
    conpleted: boolean;
}

type TodoPreview = Pick<Todo, 'title' | 'conpleted'>

const todo: TodoPreview = {
    title: 'Clean room',
    conpleted: false
}

//omit

interface Todo2 {
    title: string;
    description: string;
    conpleted: boolean;
}

type TodoPreview2 = Omit<Todo2, 'description'>

const todo2: TodoPreview2 = {
    title: 'Clean room',
    conpleted: false
}