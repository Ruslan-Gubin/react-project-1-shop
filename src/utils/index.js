export {categoryNameForData} from './categoryNameForData.ts';
export {InputValueData} from './InoputValueDataUtils';