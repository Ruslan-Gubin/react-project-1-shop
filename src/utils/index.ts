export {paginationCalculatorPage} from './paginationCalculatorPage';
export {InputValueData} from './InoputValueDataUtils';
export {formatterRub,formatter} from './intl-Number-Format';
export {getImgForSlider} from './getImgForSlider';
export {sortArray} from './sortArrayforSelect';
export {sumDiscount} from './sumDiscount';
export {totalSum, totalSumOldPrice} from './totalSum';
export {initSelect} from './initSelect';
export {filterImageInObject} from './filterImageInObject';
export { filterTags} from './getFirstTags';
export { helperLikesFunction} from './helperLikesFunction';
export { checkUserLoginForMenu} from './loginMenuArray';
export { checkCustomValidator, checkImagesValidator, checkAddProductValidator} from './checkCustomValidator'; 
export {  validVisaCard } from './validVisaCard'; 
export {  checkResorsUpdate, findNextLevelResource } from './checkResoursUpdate'; 
export {  dateGame } from './dateForGameFormat'; 
export {  incomTotalCount } from './incomTotalCount'; 
export {  calculateResourceBarUpdate } from './calculateResourceBarUpdate'; 
export {  providesList } from './providesList'; 
export {  maxValueInObjects, checkTimeUpdate } from './checkMaxTimeForResource'; 
export {  inventoryActiveFilter } from './inventoryActiveFilter'; 
export {  dragUpdateArray } from './dragUpdateArray'; 
export {  modalActiveAccordion } from './modalActiveAccordion'; 
export {  levelCheckNextUp } from './levelCheckNextUp'; 