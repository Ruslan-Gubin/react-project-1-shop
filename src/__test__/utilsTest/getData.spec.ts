import {jest, expect,test,describe, beforeEach,beforeAll,afterEach,afterAll,
} from "@jest/globals";
import axios from 'axios';
import { getData } from "../../utils";
import { responseTestData } from "../data";

jest.mock('axios')
const mockedAxios = axios as jest.Mocked<typeof axios>;

describe("Получаем данные с jsonplaceholder", () => {
  let response: any;
  
  /**Выполнится перед каждым тестом */
  beforeEach(() => {
    response = responseTestData
  });
  /**Выполняется один раз перед всеми тестами*/
  beforeAll(() => {});
  
  test("Коррекное значение", async () => {
    mockedAxios.get.mockReturnValue(response)
    const data = await getData()
    expect(axios.get).toBeCalledTimes(1);
    expect(data).toEqual(["1", "2", '3'])
    expect(data).toMatchSnapshot()
  });

  /**Выполнится после каждого теста */
  afterEach(() => {
  });
  /**Выполнится после всех тестов */
  afterAll(() => {
  });
});