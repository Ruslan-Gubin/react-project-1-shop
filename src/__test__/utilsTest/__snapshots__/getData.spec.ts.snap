// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Получаем данные с jsonplaceholder Коррекное значение 1`] = `
[
  "1",
  "2",
  "3",
]
`;
